go/vt/wrangler/keyspace.go

```diff
@@ -126,49 +126,9 @@ func (wr *Wrangler) MigrateServedTypes(ctx context.Context, keyspace, shard stri
 		return fmt.Errorf("Shard %v is not involved in any overlapping shards", shard)
 	}
 
-	// find which list is which: the sources have no source
-	// shards, the destination have source shards. We check the
-	// first entry in the lists, then just check they're
-	// consistent
-	var sourceShards []*topo.ShardInfo
-	var destinationShards []*topo.ShardInfo
-	if len(os.Left[0].SourceShards) == 0 {
-		if len(os.Right[0].SourceShards) == 0 {
-			return fmt.Errorf("neither Shard '%v' nor Shard '%v' have a 'SourceShards' entry. Did you successfully run vtworker SplitClone before? Or did you already migrate the MASTER type?", os.Left[0].ShardName(), os.Right[0].ShardName())
-		}
-		sourceShards = os.Left
-		destinationShards = os.Right
-	} else {
-		sourceShards = os.Right
-		destinationShards = os.Left
-	}
-
-	// Verify the sources has the type we're migrating (or not if reverse)
-	for _, si := range sourceShards {
-		if err := si.CheckServedTypesMigration(servedType, cells, !reverse); err != nil {
-			return err
-		}
-	}
-
-	// Verify the destinations do not have the type we're
-	// migrating (or do if reverse)
-	for _, si := range destinationShards {
```

Comment:
These well-meaning validations checked for inconsistencies in the sharding info and failed the Migrate if something's not right. However, if a Migrate fails in the middle, then the sharding info will be inconsistent, and these checks won't allow you to retry them because it will just fail saying that things are inconsistent.

Removing these checks is a big part of the idempotence fix, because this allows Migrate to continue where it left off the nest time you try.

I'm not sure how to write comments about deleted code that was unnecessary. Any suggestions?

Reply:
Ah - I meant just comments in the PR, so we can have the context here.

Reply:
Makes sense. I updated the PR with the explanation.